src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
I'm not sure I understand the issue here. Every major C library has support for custom memory allocators, and they're implemented just like we did here for CMark. Zlib is just one of thousand examples. http://www.zlib.net/manual.html

This is not a "performance trick", this is a compatibility requirement to use C libraries in real production systems, where the existing platform dictates your allocation behavior. Libraries like CMark need to be wrapped in runtimes and virtual machines that are complex, and most of the time, it's a hard requirement for the library to use the same memory allocator as the rest of the system. As an example, you probably want to use Ruby's memory allocator so the VM can track memory pressure on allocations coming from CMark, or you want to use the CLR's internal allocator to prevent having two split allocation pools inside a .NET application.

I'm well aware it adds complexity to the library, but making portable C code is really hard, whilst making C code that cannot run everywhere is easy but really not useful at all. It's a price we are paying. And I know it's making this PR more complicated for you, but there's isn't much I can do to alleviate the pain. The design we came up with is as clean as we could possibly wire it up.

I'd certainly welcome suggestions on alternate takes for allocator portability, but as @nwellnhof already pointed out, a global allocation setter is a recipe for disaster on large applications that can potentially use the same library from two different contexts. That's why we set the allocator per-parser instead of globally.